tests/conftest.py

```diff
@@ -33,6 +35,18 @@ def shared_setup(module_isolation):
     pass
 
 
+@pytest.fixture(scope="module", autouse=True)
+def reset_chain(chain):
```

Comment:
What's this used for? I didn't see it used in your other code changes, but maybe I missed it.

Reply:
resets the chain after each module